tests/evm/test_extcodesize.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+from itertools import chain
+from zkevm_specs.evm import (
+    AccountFieldTag,
+    Block,
+    Bytecode,
+    CallContextFieldTag,
+    ExecutionState,
+    RWDictionary,
+    StepState,
+    Tables,
+    verify_steps,
+)
+from zkevm_specs.util import (
+    EMPTY_CODE_HASH,
+    EXTRA_GAS_COST_ACCOUNT_COLD_ACCESS,
+    GAS_COST_WARM_ACCESS,
+    RLC,
+    U160,
+    U256,
+    keccak256,
+    rand_address,
+    rand_bytes,
+    rand_fq,
+    rand_range,
+    rand_word,
+)
+
+TESTING_DATA = [
+    (0x30000, bytes(), 0, True, True),  # warm empty account
+    (0x30000, bytes(), 0, False, True),  # cold empty account
+    (0x30000, bytes([10, 40]), 1, True, True),  # warm non-empty account
+    (0x30000, bytes([10, 10, 40]), 1, False, True),  # cold non-empty account
+    (0x30000, bytes(), 1, False, True),  # non-empty account with empty code
+    (
+        rand_address(),
+        rand_bytes(100),
+        rand_range(2),
+        rand_range(2) == 0,
+        True,  # persistent call
+    ),
+    (
+        rand_address(),
+        rand_bytes(100),
+        rand_range(2),
+        rand_range(2) == 0,
+        False,  # reverted call
+    ),
+]
+
+
+@pytest.mark.parametrize("address, code, exists, is_warm, is_persistent", TESTING_DATA)
+def test_extcodesize(address: U160, code: bytes, exists: int, is_warm: bool, is_persistent: bool):
+    randomness = rand_fq()
+
+    code_hash = int.from_bytes(keccak256(code), "big")
+    code_size = len(code) if exists == 1 else 0
+
+    tx_id = 1
+    call_id = 1
+
+    # 7 + 1 reversible operation (the account access list write)
```

Comment:
I think you can only consider successful state if fail , we will handle it in error circuit